GCEm/__init__.py

```diff
@@ -127,26 +127,75 @@ def _get_gpflow_kernel(names, n_params, active_dims=None, operator='add'):
         'mul': mul
     }
 
+    if operator not in operator_dict:
+        raise ValueError("Invalid operator: {}. Use either 'add' or 'mul'.".format(operator))
+
     def init_kernel(k):
         """
         Initialise a GPFlow kernel with the correct shape (default) variance and lengthscales
         """
-        if issubclass(k, gpflow.kernels.Constant):  # E.g., Bias
-            return k(active_dims=active_dims)
-        elif issubclass(k, gpflow.kernels.Linear):  # This covers polynomial too
-            return k(variance=[1.] * n_params, active_dims=active_dims)
-        elif issubclass(k, gpflow.kernels.Stationary):  # This covers e.g. RBF
-            return k(lengthscales=[1.] * n_params, active_dims=active_dims)
-        else:
-            raise ValueError("Invalid Kernel: {}".format(k))
+        try:
+            K_Class = kernel_dict[k]
+        except KeyError:
+            raise ValueError("Invalid Kernel: {}. Please choose from one of: {}".format(k, kernel_dict))
 
-    return reduce(operator_dict[operator], (init_kernel(kernel_dict[k]) for k in names))
+        if issubclass(K_Class, gpflow.kernels.Constant):  # E.g., Bias
+            return K_Class(active_dims=active_dims)
+        elif issubclass(K_Class, gpflow.kernels.Linear):  # This covers polynomial too
+            return K_Class(variance=[1.] * n_params, active_dims=active_dims)
+        elif issubclass(K_Class, gpflow.kernels.Stationary):  # This covers e.g. RBF
+            return K_Class(lengthscales=[1.] * n_params, active_dims=active_dims)
```

Comment:
Should have mentioned before, the White kernel doesn't sit under any of these classes. It's part of gpflow.kernels.Static, so we'll need another elif here.

The white kernel just adds a sigma^2 term along the diagonal of the kernel matrix (which is then learnt from the data) and so you don't need to define any lengthscales or the shape etc.

Reply:
Good catch - thanks!